sdv/sampling/hierarchical_sampler.py

```diff
@@ -101,10 +91,7 @@ def _sample_rows(self, synthesizer, table_name, num_rows=None):
                 Sampled rows, shape (, num_rows)
         """
         num_rows = num_rows or synthesizer._num_rows
-        if synthesizer._model:
-            sampled_rows = synthesizer._sample(num_rows)
-        else:
-            sampled_rows = pd.DataFrame(index=range(num_rows))
+        sampled_rows = synthesizer.sample(num_rows)
 
         return self._process_samples(table_name, sampled_rows)
```

Comment:
`HMA` shouldn't work if `sample` is called directly. The `synthesizers` should be fitted to the augmented `table`, which contains rows that are not recognized by the `data_processor`.
When we sample we use that data to generate the synthesizer 'state' by using `_extract_parameters` that come from this extra columns, which then are being set to a new instance of a synthesizer.

Reply:
So the issue is that `_sample` doesn't do reject sampling. This means that constraints aren't guaranteed to be met since we sometimes fall back on reject sampling. We need a way to do reject sampling but get the values back here without dropping any extra columns needed to do the algorithm

Reply:
Would it generally be the same across the multi-table sampler mixins? Could we make a base sampler mixin class and put it there?